src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AppropriationAccount.xml

```diff
@@ -1,164 +1,149 @@
+          p:label="Appropriation Account Number"
+          p:shortLabel="Approp. Acct Nbr"
+          p:maxLength="10"
+          p:validationPattern-ref="TenCharacterAlphaNumericValidation"
+          p:control-ref="TwelveCharacterTextControl">
     </bean>
-      
-  	<bean id="AppropriationAccount-projectNumber" parent="AppropriationAccount-projectNumber-parentBean"/>
-  	
-  	<bean id="AppropriationAccount-projectNumber-parentBean" abstract="true" parent="AttributeDefinition">
-    	<property name="name" value="projectNumber"/>
-    	<property name="label" value="Project Number"/>
-    	<property name="shortLabel" value="Project Number"/>
-    	<property name="maxLength" value="15"/>
-    	<property name="control">
-      		<bean parent="TextControlDefinition" p:size="15"/>
-    	</property>
-    	<property name="validationPattern">
-      		<bean parent="AlphaNumericValidationPattern">
-      			<property name="maxLength" value="15"/>
-      		</bean>
-    	</property>
-  </bean>
-  
- 
-  <bean id="AppropriationAccount-active" parent="AppropriationAccount-active-parentBean"/>
-  <bean id="AppropriationAccount-active-parentBean" abstract="true" parent="GenericAttributes-activeIndicator">
-    <property name="name" value="active"/>
-  </bean>  
 
-	
-	<!-- Business Object Inquiry Definition -->
+    <bean id="AppropriationAccount-appropriationAccountName" parent="AppropriationAccount-appropriationAccountName-parentBean"/>
+    <bean id="AppropriationAccount-appropriationAccountName-parentBean" abstract="true" parent="AttributeDefinition"
+          p:name="appropriationAccountName"
+          p:label="Appropriation Account Name"
+          p:shortLabel="Approp. Acct Name"
+          p:maxLength="40"
+          p:validationPattern-ref="AnyCharacterWithWhitespaceValidation"
+          p:control-ref="FortyTwoCharacterTextControl"/>
 
+    <bean id="AppropriationAccount-subFundGroupCode" parent="AppropriationAccount-subFundGroupCode-parentBean"/>
+    <bean id="AppropriationAccount-subFundGroupCode-parentBean" abstract="true" parent="SubFundGroup-subFundGroupCode"
+          p:shortLabel="SubFundGrpCd"/>
+
+    <bean id="AppropriationAccount-projectNumber" parent="AppropriationAccount-projectNumber-parentBean"/>
+    <bean id="AppropriationAccount-projectNumber-parentBean" abstract="true" parent="AttributeDefinition"
+          p:name="projectNumber"
+          p:label="Project Number"
+          p:shortLabel="Project Number"
+          p:maxLength="15"
+          p:validationPattern-ref="FifteenCharacterAlphaNumericValidation"
+          p:control-ref="FifteenCharacterTextControl">
+    </bean>
+
+    <bean id="AppropriationAccount-active" parent="AppropriationAccount-active-parentBean"/>
+    <bean id="AppropriationAccount-active-parentBean" abstract="true" parent="GenericAttributes-activeIndicator"
+          p:name="active"
+          p:control-ref="GenericAttributes-genericBooleanYNBoth-lookupControl">
```

Comment:
Regarding your question about active-flag terminology, it looks like older-style lookups generally use the "Active Indicator: Yes/No/Both" setup, whereas newer-style lookups generally use the "Status: Active/Inactive/All" setup. I would suggest adjusting these lookups to follow the appropriate terminology, based on whether the beans are for the old or new lookup frameworks.

I also added a related note to the Confluence instructions.